File: App/sympho-mobile-app-flutter/lib/classes/tts.dart
import "dart:typed_data";

class TTSRequest {
  final String text;
  final String voiceId;
  final TTSVoiceSettings voiceSettings;

  TTSRequest({
    required this.text,
    required this.voiceId,
    required this.voiceSettings,
  });

  Map<String, dynamic> toMap() {
    return {
      'text': text,
      'voiceId': voiceId,
      "voiceSettings": voiceSettings.toMap(),
    };
  }
}

class TTSVoiceSettings {
  final double stability;
  final double similarityBoost;
  final double style;
  final bool useSpeakerBoost;
  final double speed;

  TTSVoiceSettings({
    required this.stability,
    required this.similarityBoost,
    required this.style,
    required this.useSpeakerBoost,
    required this.speed,
  });

  Map<String, dynamic> toMap() {
    return {
      'stability': stability,
      'similarityBoost': similarityBoost,
      'style': style,
      'useSpeakerBoost': useSpeakerBoost,
      'speed': speed,
    };
  }
}

class TTSResponse {
  Uint8List? audio;
  String? error;

  TTSResponse({this.audio, this.error});
}

File: API/sympho-api-rust/src/structs/request/tts.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct TTSBody {
    pub text: String,
    pub voice_id: String,
    pub voice_settings: ClientVoiceSettings,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ClientVoiceSettings {
    pub stability: f32,
    pub similarity_boost: f32,
    pub style: f32,
    pub use_speaker_boost: bool,
    pub speed: f32,
}
